packages/edit-site/src/components/sidebar/global-styles-sidebar.js

```diff
@@ -30,122 +35,116 @@ import {
 } from './dimensions-panel';
 
-function GlobalStylesPanel( {
-	wrapperPanelTitle,
+function getPanelTitle( blockName ) {
+	const blockType = getBlockType( blockName );
+
+	// Protect against blocks that aren't registered
+	// eg: widget-area
+	if ( blockType === undefined ) {
+		return blockName;
+	}
+
+	return blockType.title;
+}
+
+function GlobalStylesLevel( {
 	context,
 	getStyle,
 	setStyle,
 	getSetting,
 	setSetting,
+	parentMenu = 'root',
 } ) {
-	const hasBorderPanel = useHasBorderPanel( context );
-	const hasColorPanel = useHasColorPanel( context );
 	const hasTypographyPanel = useHasTypographyPanel( context );
+	const hasColorPanel = useHasColorPanel( context );
+	const hasBorderPanel = useHasBorderPanel( context );
 	const hasDimensionsPanel = useHasDimensionsPanel( context );
+	const hasLayoutPanel = hasBorderPanel || hasDimensionsPanel;
 
-	if ( ! hasColorPanel && ! hasTypographyPanel && ! hasDimensionsPanel ) {
-		return null;
-	}
-
-	const content = (
+	return (
 		<>
 			{ hasTypographyPanel && (
-				<TypographyPanel
-					context={ context }
-					getStyle={ getStyle }
-					setStyle={ setStyle }
+				<NavigationItem
+					item="item-typography"
+					navigateToMenu={ parentMenu + '.typography' }
+					title={ __( 'Typography' ) }
 				/>
 			) }
 			{ hasColorPanel && (
-				<ColorPanel
-					context={ context }
-					getStyle={ getStyle }
-					setStyle={ setStyle }
-					getSetting={ getSetting }
-					setSetting={ setSetting }
-				/>
-			) }
-			{ hasDimensionsPanel && (
-				<DimensionsPanel
-					context={ context }
-					getStyle={ getStyle }
-					setStyle={ setStyle }
+				<NavigationItem
+					item="item-color"
+					navigateToMenu={ parentMenu + '.color' }
+					title={ __( 'Colors' ) }
 				/>
 			) }
-			{ hasBorderPanel && (
-				<BorderPanel
-					context={ context }
-					getStyle={ getStyle }
-					setStyle={ setStyle }
+			{ hasLayoutPanel && (
+				<NavigationItem
+					item="item-layout"
+					navigateToMenu={ parentMenu + '.layout' }
+					title={ __( 'Layout' ) }
 				/>
 			) }
-		</>
-	);
-	if ( ! wrapperPanelTitle ) {
-		return content;
-	}
-	return (
-		<PanelBody title={ wrapperPanelTitle } initialOpen={ false }>
-			{ content }
-		</PanelBody>
-	);
-}
 
-function getPanelTitle( blockName ) {
-	const blockType = getBlockType( blockName );
-
-	// Protect against blocks that aren't registered
-	// eg: widget-area
-	if ( blockType === undefined ) {
-		return blockName;
-	}
+			{ hasTypographyPanel && (
+				<NavigationMenu
+					menu={ parentMenu + '.typography' }
+					parentMenu={ parentMenu }
+				>
+					<NavigationItem>
+						<TypographyPanel
+							context={ context }
+							getStyle={ getStyle }
+							setStyle={ setStyle }
+						/>
+					</NavigationItem>
+				</NavigationMenu>
+			) }
 
-	return blockType.title;
-}
+			{ hasColorPanel && (
+				<NavigationMenu
+					menu={ parentMenu + '.color' }
+					parentMenu={ parentMenu }
+				>
+					<NavigationItem>
+						<ColorPanel
```

Comment:
Just because of wrapping our current components with the navigation system the design becomes broken:
![image](https://user-images.githubusercontent.com/11271197/133669043-b7d5abed-9b43-4e84-a4d0-dfd56b32d06c.png)
![image](https://user-images.githubusercontent.com/11271197/133669142-9cd943fb-dcdb-4d2e-b120-2e842b77dde8.png)


I guess the styles of the navigation-related components are leaking and it was an already existing problem that we need to fix?

Reply:
yeah, I noticed this, we need to find a way to make the navigation styles more specific to the navigation components